src/main/java/org/opensearch/ad/rest/RestSearchAnomalyDetectorInfoAction.java

```diff
@@ -17,36 +17,59 @@
 import java.io.IOException;
 import java.util.List;
 import java.util.Locale;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.opensearch.ad.AnomalyDetectorPlugin;
+import org.opensearch.action.ActionListener;
+import org.opensearch.ad.AnomalyDetectorExtension;
 import org.opensearch.ad.constant.CommonErrorMessages;
+import org.opensearch.ad.settings.AnomalyDetectorSettings;
 import org.opensearch.ad.settings.EnabledSetting;
 import org.opensearch.ad.transport.SearchAnomalyDetectorInfoAction;
 import org.opensearch.ad.transport.SearchAnomalyDetectorInfoRequest;
-import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.ad.transport.SearchAnomalyDetectorInfoResponse;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.extensions.rest.ExtensionRestResponse;
 import org.opensearch.rest.RestHandler;
 import org.opensearch.rest.RestRequest;
-import org.opensearch.rest.action.RestToXContentListener;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.sdk.ExtensionsRunner;
+import org.opensearch.sdk.SDKClient.SDKRestClient;
+import org.opensearch.sdk.rest.BaseExtensionRestHandler;
+import org.opensearch.sdk.rest.ReplacedRouteHandler;
 
 import com.google.common.collect.ImmutableList;
 
-public class RestSearchAnomalyDetectorInfoAction extends BaseRestHandler {
+public class RestSearchAnomalyDetectorInfoAction extends BaseExtensionRestHandler {
 
     public static final String SEARCH_ANOMALY_DETECTOR_INFO_ACTION = "search_anomaly_detector_info";
 
     private static final Logger logger = LogManager.getLogger(RestSearchAnomalyDetectorInfoAction.class);
+    private SDKRestClient client;
+    private ExtensionsRunner extensionsRunner;
 
-    public RestSearchAnomalyDetectorInfoAction() {}
+    public RestSearchAnomalyDetectorInfoAction(ExtensionsRunner extensionsRunner, SDKRestClient client) {
```

Comment:
Not a blocker for this PR: Curious if there is a better way to get information from extensions runner. 
We just dont need to pass this across the extension.

Reply:
That's a good question. @dbwiddis do you think we can add ExtensionsRunner in createComponent and inject from there?

Reply:
`ExtensionsRunner `is already bound to guice [here](https://github.com/opensearch-project/opensearch-sdk-java/blob/2991b5f3f2bd1fc34413134036056b3f518c3063/src/main/java/org/opensearch/sdk/ExtensionsRunner.java#L219). I think what we need is to enable injection for rest actions so that we don't need to pass objects around in AD Extension